### PR TITLE
Add `Viewer` plugin support 

### PR DESCRIPTION
**Pull request by wykhuh**
@adamjarling @mathewjordan These are changes I made Clover in order to support plugins. The two parts that I changed for the annotation plugin was to give the plugin access to the Image Viewer Controls and Information Panel.

### CODE REVIEW

**Comment by wykhuh** [comment]
@adamjarling @mathewjordan I updated plugins to Clover 2.7.5.

**Review by mathewjordan — COMMENTED** [review]
🥇 Overall, this looks pretty good to me. I'll chat through it with @adamjarling now that I grok it well enough. If there are changes, I think they will be pretty minimal and mostly organizational.

File: src/types/plugins.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export interface PluginInformationPanel {
+  annotations?: Annotation[];
+  activeManifest: string;
+  canvas: CanvasNormalized;
+  viewerConfigOptions: ViewerConfigOptions;
+  openSeadragonViewer: OpenSeadragon.Viewer | null;
+  useViewerDispatch: () => ViewerContextStore;
+  useViewerState: () => ViewerContextStore;
+}
```
Comment (adamjarling) [inline src/types/plugins.ts:24]: Maybe `extend` PluginInformationPanel?

```ts
export interface PluginInformationPanel extends Plugin {
  annotations?: Annotation[];
}
```

File: src/index.tsx [hunk 2]
```diff
@@ -3,7 +3,24 @@ import Primitives from "src/components/Primitives";
...
+  Image,
+  Primitives,
+  Scroll,
+  Slider,
+  Viewer,
+  parseAnnotationTarget,
+  type AnnotationTargetExtended,
+  createOpenSeadragonRect,
+  type Plugin,
+  type PluginInformationPanel,
```
Comment (adamjarling) [inline src/index.tsx:23]: I also wonder if there's cleaner way to handle these exports, or group or title them to clarify their context to future consumers of the package?

**Comment by wykhuh** [comment]
> @wykhuh @mathewjordan Should we add some documentation?

I was going to add documentation after y'all looked the PR. I wanted us to figure out the api for plugins before writing the documentations.

Another thing to think about: will Clover website have a page that lists the custom displays and plugins that are available for Clover?

**Comment by wykhuh** [comment]
@mathewjordan @adamjarling You two have edit access to my forked repo, so feel free to make edits and commits to this branch.

File: src/context/viewer-context.tsx [hunk 3]
```diff
@@ -138,6 +141,20 @@ export type CustomDisplay = {
     paintingFormat?: string[];
   };
 };
+export type PluginConfig = {
+  id: string;
+  imageViewer?: {
+    menu?: {
```
Comment (mathewjordan) [inline src/context/viewer-context.tsx:147]:
```suggestion
    controls?: {
```

File: src/types/plugins.ts (hunk 1 above)
Comment (mathewjordan) [reply src/types/plugins.ts:24]: Agree here.

**Review by mathewjordan — CHANGES_REQUESTED** [review]
A few notes:

- Add plugins docs in Viewer
- Consolidate helpers under one export 

```
  'helpers': {
    lib: {
      name: "CloverIIIFHelpers",
      entry: "./src/lib/index.ts", // pile them through here
      fileName: "index",
    },
  }
```

- Rename `imageViewer.menu` to `imageViewer.controls`
- If you'd like, add a Plugins page under Composing in the documentation nav
- Adam's note about extending types where possible

File: build/build.mjs [hunk 4]
```diff
@@ -38,6 +38,20 @@ const buildOptions = {
...
+      name: "CloverIIIFAnnotationHelpers",
+      entry: "./src/lib/annotation-helpers.ts",
+      fileName: "index",
+    },
+  },
+  'openseadragon-helpers': {
+    lib: {
+      name: "CloverIIIFOpenseadragonHelpers",
+      entry: "./src/lib/openseadragon-helpers.ts",
+      fileName: "index",
```
[the remark this answers is not in the document]
Comment (wykhuh) [reply build/build.mjs]: I consolidated the helpers in a single export

File: src/context/viewer-context.tsx (hunk 3 above)
Comment (wykhuh) [reply src/context/viewer-context.tsx]: I changed it to imageViewer.controls

**Comment by wykhuh** [comment]
@mathewjordan @adamjarling I mage the suggested changes. I also add plugin docs to the viewer page. Could you review the docs to see if it makes sense?

**Comment by mathewjordan** [comment]
> @mathewjordan @adamjarling I mage the suggested changes. I also add plugin docs to the viewer page. Could you review the docs to see if it makes sense?

Sure thing.

File: src/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (wykhuh) [reply src/index.tsx]: @mathewjordan I exported a consolidated helpers.

**Review by mathewjordan — APPROVED** [review]

**Pull request merged**